jsonnet/kube-prometheus/components/prometheus-adapter.libsonnet

```diff
@@ -162,14 +163,8 @@ function(params) {
         labels: pa._config.commonLabels,
       },
       spec: {
-        replicas: 1,
+        replicas: pa._config.replicas,
         selector: { matchLabels: pa._config.selectorLabels },
-        strategy: {
-          rollingUpdate: {
-            maxSurge: 1,
-            maxUnavailable: 0,
-          },
-        },
```

Comment:
Removing those can cause downtime during upgrades. This is because by default both parameters are set to 25% and k8s is rounding numbers down.

Reply:
Good catch! It was indeed wrong to completely remove it as the default rolling-update strategy doesn't prevent any downtime. I amended the original configuration and changed maxUnavailable to 1 as not allowing unavailable replicas might cause scheduling issues with hard-affinity constraints when the number of replicas is equal to the number of available nodes.